sunkit_dem/base_model.py

```diff
@@ -110,24 +112,22 @@ def kernel(self):
     def kernel(self, kernel):
         if len(kernel) != self.data.cube_like_dimensions[0].value:
             raise ValueError('Number of kernels must be equal to length of wavelength dimension.')
-        if not all([kernel[k].shape == self.temperature_bin_centers.shape for k in kernel]):
+        if not all([k.shape == self.temperature_bin_centers.shape for k in kernel]):
             raise ValueError('Temperature and kernels must have the same shape.')
         self._kernel = kernel
 
     @property
     def wavelength(self):
         # TODO: handle non-numeric "wavelength" designations
-        return u.Quantity(self.data.common_axis_extra_coords['wavelength'],
-                          self.data[0].wcs.wcs.cunit[-1])
+        return u.Quantity(self.data.common_axis_coords[0], self.data[0].wcs.wcs.cunit[-1])
 
     @property
     def data_matrix(self):
         return u.Quantity(np.stack([n.data.squeeze() for n in self.data]), self.data[0].unit)
 
     @property
     def kernel_matrix(self):
-        return u.Quantity(np.stack([self.kernel[w].value for w in self.wavelength.value]),
-                          self.kernel[self.wavelength.value[0]].unit)
+        return u.Quantity(np.stack([k.value for k in self.kernel]), self.kernel[0].unit)
```

Comment:
As with my above comment, I also had to change this back to:

```

    @property
    def kernel_matrix(self):
        first_key = list(self.kernel)[0]
        return u.Quantity(
            np.stack([self.kernel[k].value for k in self.kernel]),
            self.kernel[first_key].unit,
        )
```

Reply:
I think this should now be fixed as I'm constructing the kernel matrix a bit differently and being careful to preserve the correct ordering between the data matrix and the kernel matrix.